### PR TITLE
Guess movie poster link from event description

### PR DESCRIPTION
**Pull request by meo84**
-- What? 
Simplify the form that creates a new movie: when the user selects a meetup event in the form, the link of the movie poster in the form is automatically filled out. 

-- How? 
I used an AJAX call on the `/movies/new` page, but wasn't sure how to test the call itself/the view, so only did some unit tests on the methods I used from the `Event` model to support the call (`.summary`, `#preview`, `#image_urls`, `#first_image_url`).

-- Questions?
If you have some suggestions on how I could test the AJAX call, or some feedback on the call itself and the other tests, they'll all be welcome!

### CODE REVIEW

File: Gemfile [hunk 1]
```diff
@@ -37,6 +37,9 @@ gem 'countries'
 #select country info in the ISO standard
 gem 'country_select'
 
+#parse html
+gem 'nokogiri', '~>1.6.8'
```
Comment (yaf) [inline Gemfile]: Add gem versions in `Gemfile` seems a duplication with `Gemfile.lock`. Not a good idea. Remove them until they are absolutly necessary.

File: Gemfile [hunk 2]
```diff
@@ -46,8 +49,9 @@ gem 'country_select'
 group :development, :test do
   # Call 'byebug' anywhere in the code to stop execution and get a debugger console
   gem 'byebug'
-  gem 'rspec-rails'
-  gem 'shoulda'
+  gem 'rspec-rails', '~>3.5.2'
+  gem 'shoulda', '~>3.5.0'
+  gem 'factory_girl_rails', '~>4.8.0'
```
Comment (yaf) [inline Gemfile]: Same here about Gem version number

File: app/assets/javascripts/movies.js [hunk 3]
```diff
@@ -0,0 +1,19 @@
+$(function() {
+  $('html').on('change', 'select#movie_event_ids.chosen', function(){
```
Comment (yaf) [inline app/assets/javascripts/movies.js]: I don't look at your app in production, but `change` event could be fired a lot (focus on select field, use keyboard arrow, on worst, slide with pad...). Maybe a little button or a timer could help ?

File: app/assets/javascripts/movies.js [hunk 4]
```diff
@@ -0,0 +1,19 @@
...
+
+    $.ajax({
+      type: "GET",
+      url: form.attr('action') + "/new",
+      success: function(data){
+        var chosenEventID = $('#movie_event_ids').val();
+        var chosenEvent = data.find(function(event){ return event["id"] == chosenEventID; });
+        var chosenEventImageURL = chosenEvent["first_image_url"];
+        $('#movie_poster_url').val(chosenEventImageURL);
+      },
```
Comment (yaf) [inline app/assets/javascripts/movies.js]: Often, `get` ajax call return a piece of html that you could use to change a piece of your DOM... Could it be a better idea here ?

**Review by yaf — CHANGES_REQUESTED** [review]
I think you must remove Gem version number of your Gemfile, and change some stuff around you Ajax call.

File: Gemfile (hunk 1 above)
Comment (meo84) [reply Gemfile]: When I remove the gem's version numbers from the Gemfile and run `bundle install`, it also removes the gem's version numbers from the Gemfile.lock. How can I know the versions I used later on?

File: app/assets/javascripts/movies.js (hunk 3 above)
Comment (meo84) [reply app/assets/javascripts/movies.js]: I'm not sure I understand, I used event delegation with the [.on](https://api.jquery.com/on/) method to make sure the `change` event is only fired when the `select#movie_event_ids.chosen` element is changed. 
Can the event be fired even if I don't see requests on the network console?

File: Gemfile (hunk 1 above)
Comment (yaf) [reply Gemfile]: I don't think so, `Gemfile.lock` keep version of each installed gem.

File: app/assets/javascripts/movies.js (hunk 3 above)
Comment (yaf) [reply app/assets/javascripts/movies.js]: Try to scroll over your select box and look if your JavaScript keep calm or become mad

File: Gemfile (hunk 1 above)
Comment (meo84) [reply Gemfile]: Thanks, you're right, I read it too fast and jumped to the dependencies section 🙊

File: app/assets/javascripts/movies.js (hunk 3 above)
Comment (meo84) [reply app/assets/javascripts/movies.js]: JS seems to be keeping zen :)

File: app/assets/javascripts/movies.js (hunk 4 above)
Comment (meo84) [reply app/assets/javascripts/movies.js]: Good idea, thanks!

**Review by yaf — APPROVED** [review]

**Pull request merged**